backend/gke/deploy.sh

```diff
@@ -39,16 +39,33 @@ cluster_name   = "$GKE_CLUSTER_NAME"
 k8s_version    = "latest"
 HEREDOC
 
+if [ -n "${TF_KEY}" ] ; then
+    cat > backend.tf <<EOF
+terraform {
+  backend "s3" {
+      bucket = "${TF_BUCKET}"
+      region = "${TF_REGION}"
+      key    = "${TF_KEY}"
+  }
+}
+EOF
+fi
+
 # terraform needs helm client installed and configured:
 helm_init_client
 
 info "Deploying GKE cluster with terraform…"
 
 terraform init
 
-terraform plan -out="$(pwd)"/my-plan
+terraform plan -out=my-plan
+
+if [ -n "${TF_KEY}" ] ; then
+    zip -r9  "${BUILD_DIR}/tf-setup.zip" .
```

Comment:
I am not sure exactly what's being done here.

Reply:
zips the file for concourse pool. Same as in `master` for EKS: https://github.com/SUSE/catapult/blob/master/backend/eks/deploy.sh#L50-L51